protocol/bc/nonce.go

```diff
@@ -42,3 +47,27 @@ func (n *Nonce) SetAnchored(id Hash, entry Entry) {
 	n.Witness.AnchoredID = id
 	n.Anchored = entry
 }
+
+func (n *Nonce) checkValid(vs *validationState) error {
```

Comment:
Here's what your suggestion looks like: https://github.com/chain/chain/pull/788/commits/425e060fa45136f6ee032630bbd9f95fadb24ffe. Keep it? Revert it? Change it?